cluster-autoscaler/cloudprovider/aws/ec2_instance_types/gen.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +build ignore
+
+package main
+
+import (
+	"encoding/json"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/golang/glog"
+)
+
+type response struct {
+	Products map[string]product `json:"products"`
+}
+
+type product struct {
+	Attributes productAttributes `json:"attributes"`
+}
+
+type productAttributes struct {
+	InstanceType string `json:"instanceType"`
+	VCPU         string `json:"vcpu"`
+	Memory       string `json:"memory"`
+}
+
+type instanceType struct {
+	InstanceType string
+	VCPU         int64
+	Memory       int64
+}
+
+var packageTemplate = template.Must(template.New("").Parse(`/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file was generated by go generate; DO NOT EDIT
+
+package aws
+
+type instanceType struct {
+	InstanceType string
+	VCPU         int64
+	MemoryMb     int64
+}
+
+// InstanceTypes is a map of ec2 resources
+var InstanceTypes = map[string]*instanceType{
+{{- range .InstanceTypes }}
+	"{{ .InstanceType }}": {
+		InstanceType: "{{ .InstanceType }}",
+		VCPU:         {{ .VCPU }},
+		MemoryMb:     {{ .Memory }},
+	},
+{{- end }}
+}
+`))
+
+func main() {
+	instanceTypes := make(map[string]*instanceType)
```

Comment:
Ah, or is this an intended behavior, assuming all the instance types named the same to have equally sized cpu and memory across different regions?